src/structure/structure.ts

```diff
@@ -123,10 +128,10 @@ export class StructureViewer {
 
     // GUID of the Active Widget
     private _active: string;
+    // Map of Widgets GUIDS to their color, widget, and current structure
+    private _widgetMap: {[GUID: string]: GriddedWidget; };
```

Comment:
An explicit `Map` might be better here, since we also want to delete elements from it. What you've written here is a JavaScript object with unknown keys, which is slightly different (see Object vs Map in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map)